Services/Navigation/classes/class.ilNavigationHistory.php

```diff
@@ -131,37 +131,33 @@ public function getItems() : array
             }
         }
         // less than 10? -> get items from db
-        if (count($items) < 10 && $ilUser->getId() != ANONYMOUS_USER_ID) {
+        if (count($items) < 10 && $ilUser->getId() !== ANONYMOUS_USER_ID) {
             $set = $ilDB->query(
                 "SELECT last_visited FROM usr_data " .
                 " WHERE usr_id = " . $ilDB->quote($ilUser->getId(), "integer")
             );
             $rec = $ilDB->fetchAssoc($set);
-            $db_entries = unserialize($rec["last_visited"]);
+            $db_entries = unserialize($rec["last_visited"], ["allowed_classes" => false]);
             $cnt = count($items);
             if (is_array($db_entries)) {
                 foreach ($db_entries as $rec) {
-                    if ($cnt <= 10 && !isset($items[$rec["ref_id"] . ":" . $rec["sub_obj_id"]])) {
-                        if (
-                            $tree->isInTree($rec["ref_id"]) &&
-                            (
-                                !$objDefinition->isPluginTypeName($rec["type"]) ||
-                                $component_repository->getPluginById($it["type"])->isActive()
-                            )
-                        ) {
-                            $link = ($rec["goto_link"] != "")
+                    if ($cnt <= 10 && !isset($items[$rec["ref_id"] . ":" . $rec["sub_obj_id"]]) && $tree->isInTree($rec["ref_id"]) &&
+                        (
+                            !$objDefinition->isPluginTypeName($rec["type"]) ||
+                            $component_repository->getPluginById($it["type"])->isActive()// TODO PHP8-REVIEW $it is based on the laste iteration above. I suggest to use an explicit new variable for this case.
```

Comment:
`$it` is based on the laste iteration above. I suggest to use an explicit new variable for this case

Reply:
This made me puzzling, its a wrong variable being accessed. But a trunk only issue, see https://github.com/ILIAS-eLearning/ILIAS/commit/4786c2e55dda9532240ed3510480eb08dae3a684#diff-c3c5149c3990e61a2f036839a286269837736dff68894bb855fc6dc0a5e89090